base/linalg/arpack.jl

```diff
@@ -228,7 +228,7 @@ for (T, saupd_name, seupd_name, naupd_name, neupd_name) in
                   sizeof(howmny), sizeof(bmat), sizeof(evtype))
         end
 
-        function saupd(ido, bmat, n, which, nev, TOL::Array{$T}, resid::Array{$T}, ncv, v::Array{$T}, ldv,
+        function saupd(ido, bmat, n, which, nev, TOL::Ref{$T}, resid::Array{$T}, ncv, v::Array{$T}, ldv,
```

Comment:
How about allowing the future form with just `T` now in the 0.6 cycle? That won't break anyone's code and then people can write calls already the way it will be in the future.